syntax = "proto3";

option java_multiple_files = true;
option java_outer_classname = "Video";
option java_package = "io.grpc.videostreamer";
option objc_class_prefix = "VS";

package videostreamer;

service VideoStreamer {
    rpc VideoProcess (InputVideo) returns (OutputVideo) {}
    rpc VideoProcessFromC (InputVideo) returns (OutputVideo) {}
    rpc VideoProcessFromAndroid (stream InputVideo) returns (stream BoundingBox) {}
}

message InputVideo {
    bytes frame = 1;
}

message OutputVideo {
    bytes frame = 1;
}

message BoundingBox {
    string info = 1;
}
